plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/names/NameScope.kt

```diff
@@ -27,11 +27,25 @@ data class GlobalScope(override val packageName: FqName) : PackagePrefixScope {
     constructor(segments: List<String>) : this(FqName.fromSegments(segments))
 }
 
-data class ClassScope(override val packageName: FqName, val className: ClassKotlinName) : PackagePrefixScope {
-    // We don't need to prefix this with "class_scope" since the class name will already be prefixed with "class".
+sealed interface ClassScope : PackagePrefixScope {
+    val className: ClassKotlinName
+}
+
+data class DefaultClassScope(override val packageName: FqName, override val className: ClassKotlinName, ) : ClassScope {
     override val suffix = className.mangled
+}
+
+/**
+ * We do not want to mangle field names with class and package, hence introducing
+ * this special `NameScope`. Note that it still needs package and class for other purposes.
+ */
+data class PropertyClassScope(override val packageName: FqName, override val className: ClassKotlinName) : ClassScope {
+    override val suffix = className.mangled + "_property"
```

Comment:
nit: I think `backing_field` would be better when that's what we generate, with `property` used for getter/setter